src/gui/mzroll/background_peaks_update.cpp

```diff
@@ -586,69 +569,63 @@ void BackgroundPeakUpdate::align() {
                         mainwindow->alignmentPolyVizDockWidget->setDegreeMap(aligner.sampleDegree);
                         mainwindow->alignmentPolyVizDockWidget->setCoefficientMap(aligner.sampleCoefficient);
                 } else if (alignAlgo == 1) {
-                        aligner.preProcessing(groups);
-
+                        aligner.preProcessing(groups, mavenParameters->alignWrtExpectedRt);
+                        // initialize processedDataFromPython with null 
+                        processedDataFromPython="";
                          /**runPythonProg()
                          * sends the json of groups and samples rt to the python exe. for more look in sendDataToPython()
                          * python exe is going to correct the rts and send it back to us in json format
                          */
                         runPythonProg(&aligner);
-
-                        /**readDataFromPython(data)
-                         * it will wait for python to send the corrected rts
-                         * once we receive the corrected rts, we kill the python exe and update the rts in maven
-                         */
-                        QByteArray data;
-                        readDataFromPython(data);
+                        // wait for processing of data by python program
+                        pythonProg->waitForFinished(-1);                        
 
                         // convert the data to json
                         QJsonDocument jDoc;
                         QJsonObject parentObj;
 
                         // if jDoc is null that means the json returned from python is malformed
                         // in such a case our rts wont update with new values
-                        jDoc = QJsonDocument::fromJson(data);
-                        if(!jDoc.isNull())
-                        parentObj = jDoc.object();
-
-                        if(!parentObj.isEmpty())
-                        aligner.updateRts(parentObj);
+                        jDoc = QJsonDocument::fromJson(processedDataFromPython);
+
+                        QString errorMessage=QString::number(processedDataFromPython.size());
```

Comment:
error message creation should move into the else section, no?

Reply:
@shubhra-agrawal It's part of error message and common to two places.